charts/edge/templates/_helpers.tpl

```diff
@@ -67,3 +67,38 @@ Generate the base URL of the Mezmo API
 {{- define "edge.mezmoApiBaseUrl" -}}
 {{- printf "%s://%s" .Values.mezmoApiScheme .Values.mezmoApiHost }}
 {{- end }}
+
+{{/*
+Ports as a yaml array
+*/}}
+{{- define "edge.sourcePorts" -}}
+{{- $ports := .Values.service.sourcePorts.list }}
+{{- $start := .Values.service.sourcePorts.start | default 0 | int }}
+{{- $end := .Values.service.sourcePorts.end | default 0 | int }}
+{{- range $port := untilStep $start (add1 $end | int) 1 }}
+  {{- $ports = append $ports $port }}
+{{- end }}
+{{- $ports | uniq | toYaml}}
+{{- end }}
+
+{{/*
+Defaulting Edge ID
+*/}}
+{{- define "edge.Id" -}}
+{{- $id_hash := sha1sum (print .Release.Name .Release.Namespace .Values.mezmoDeploymentGroup) }}
```

Comment:
Shouldn't this be a `uuidv4` instead of a consistent hash?

Is there anything in Helm preventing the same release name in different deployments? i.e., different helm clients using the same release name.

Reply:
I didn't go UUID because I was favoring stability of identity. I wanted something deployed with the same (namespace, name, deployment group) to appear as the same thing in the UX, because that's how I think users will think of it. I thought this would provide good continuity for folks installing or uninstalling the same "thing" (where the "thing" goes away entirely and comes back new, but with the same config -- I additionally didn't even find a great way to make a random UUID even survive in-place upgrades, which also wouldn't lend to continuity). 

As far as basic uniqueness by default, K8s will not allow duplicate (namespace, name) within the same cluster, and I imagined the deployment group would additionally disambiguate among environments (clusters). The override is there as an escape hatch for more esoteric scenarios, but I haven't thought of any real deployment models that would break in defaults.

Another thought: we could detect "too frequent" heartbeats and indicate in the record-->UX that we might have duplicate instances running somehow.

I can revisit if this rationale does not hold water with reviewers.

Reply:
> As far as basic uniqueness by default, K8s will not allow duplicate (namespace, name) within the same cluster

I thought that Helm release name was stored on the client side but [it looks like it's stored on the K8s side](https://helm.sh/docs/topics/advanced/#storage-backends): nvm, a stable hash is better.